cmd/cvp-trigger/main.go

```diff
@@ -4,69 +4,165 @@ import (
-	o := options{}
-	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+// validateOptions ensures that all required flag options are present and
+// validates any constraints on appropriate values
+func (o options) validateOptions() error {
+	afs := afero.Afero{Fs: fileSystem}
 
-	fs.BoolVar(&o.confirm, "confirm", false, "Whether to actually submit the job to Prow")
-	fs.StringVar(&o.prowConfigPath, prowConfigPathOption, "", "Path to the Prow config file")
-	fs.StringVar(&o.jobConfigPath, jobConfigPathOption, "", "Path to the Prow job config directory")
-	fs.StringVar(&o.periodic, "periodic", "", "Name of the Periodic job to manually trigger")
+	if o.bundleImageRef == "" {
+		return fmt.Errorf("required parameter %s was not provided", bundleImageRefOption)
+	}
 
-	_ = fs.Parse(os.Args[1:])
-	return o
-}
+	if o.channel == "" {
+		return fmt.Errorf("required parameter %s was not provided", channelOption)
+	}
 
-func (o options) validate() error {
-	if o.prowConfigPath == "" {
-		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	if o.indexImageRef == "" {
+		return fmt.Errorf("required parameter %s was not provided", indexImageRefOption)
 	}
 
 	if o.jobConfigPath == "" {
 		return fmt.Errorf("required parameter %s was not provided", jobConfigPathOption)
 	}
+	exists, _ := afs.Exists(o.jobConfigPath)
+	if !exists {
+		return fmt.Errorf("validating job config path %s failed, does not exist", o.jobConfigPath)
+	}
+
+	if o.jobName == "" {
+		return fmt.Errorf("required parameter %s was not provided", jobNameOption)
+	}
+
+	if o.ocpVersion == "" {
+		return fmt.Errorf("required parameter %s was not provided", ocpVersionOption)
+	}
+	if !strings.HasPrefix(o.ocpVersion, "4") {
+		return fmt.Errorf("ocp-version must be 4.x or higher")
+	}
 
-	if o.periodic == "" {
-		return fmt.Errorf("required parameter %s was not provided", periodicOption)
+	if o.packageName == "" {
+		return fmt.Errorf("required parameter %s was not provided", packageNameOption)
+	}
+
+	if o.prowConfigPath == "" {
+		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	}
+	exists, _ = afs.Exists(o.prowConfigPath)
+	if !exists {
+		return fmt.Errorf("validating prow config path %s failed, does not exist", o.prowConfigPath)
 	}
 
 	return nil
 }
 
+// getClusterConfig returns a config from calling util.LoadClusterConfig()
+func getClusterConfig() (*rest.Config, error) {
+	var clusterConfig *rest.Config
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return clusterConfig, nil
+}
+
+// getProwJob returns a Prow Job or an error if the provided job name is not found
+func getProwJob(jobName string, config *prowconfig.Config) (*pjapi.ProwJob, error) {
+	var selectedJob *prowconfig.Periodic
+	for _, job := range config.AllPeriodics() {
+		if job.Name == jobName {
+			selectedJob = &job
+			break
+		}
+	}
+
+	if selectedJob == nil {
+		return nil, fmt.Errorf("failed to find the job")
```

Comment:
You likely want the name of the job in this error message

Reply:
Will update to include the job name in the error message, and call `util.LoadClusterConfig()` directly in main() instead.